### PR TITLE
kubeadm: improvements to the cert management documentation

### PR DESCRIPTION
**Pull request by neolit123**
- move the sections about custom certificates and external CA
to the kubeadm-certs page
- minor cleanups to the kubeadm-certs page, including updated output
for the check-expiration command
- link the implementation details page to the new locations
for custom certs and external CA

fixes kubernetes/kubeadm#1770

/kind cleanup
/priority important-longterm
/assign @fabriziopandini @jimangel

### CODE REVIEW

File: content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md [hunk 1]
```diff
@@ -177,30 +176,12 @@ The following command can be used to generate a new key on demand:
...
-It is also possible to provide just the `ca.crt` file and not the
-`ca.key` file (this is only available for the root CA file, not other cert pairs).
-If all other certificates and kubeconfig files are in place, kubeadm recognizes
-this condition and activates the "External CA" mode. kubeadm will proceed without the
-CA key on disk.
-
-Instead, run the controller-manager standalone with `--controllers=csrsigner` and
-point to the CA certificate and key.
+For detailed information on certificate management with kubeadm see
+[here](/docs/tasks/administer-cluster/kubeadm/kubeadm-certs/).
```
Comment (sftim) [inline content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md]: style nit:
```suggestion
[Certificate Management with kubeadm](/docs/tasks/administer-cluster/kubeadm/kubeadm-certs/).
```

File: content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-certs.md [hunk 2]
```diff
@@ -21,9 +21,37 @@ You should be familiar with [PKI certificates and requirements in Kubernetes](/d
...
+
+By default, kubeadm generates all the certificates needed for a cluster to run.
+You can override this behavior by providing your own certificates.
+
+To do so, you must place them in whatever directory is specified by the
+`--cert-dir` flag or the `certificatesDir` field of kubeadm's `ClusterConfiguration`.
+By default this is `/etc/kubernetes/pki`.
+
+If a given certificate and private key pair exists before running `kubeadm init`,
+kubeadm will not overwrite them. This means you can, for example, copy an existing
```
Comment (sftim) [inline content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-certs.md]: style mini-nit:
```suggestion
kubeadm does not overwrite them. This means you can, for example, copy an existing
```

File: content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-certs.md [hunk 3]
```diff
@@ -21,9 +21,37 @@ You should be familiar with [PKI certificates and requirements in Kubernetes](/d
...
+`ca.key` file (this is only available for the root CA file, not other cert pairs).
+If all other certificates and kubeconfig files are in place, kubeadm recognizes
+this condition and activates the "External CA" mode. kubeadm will proceed without the
+CA key on disk.
+
+Instead, run the controller-manager standalone with `--controllers=csrsigner` and
+point to the CA certificate and key.
+
+A more detailed guide about setting up external CA can be found
+[here](https://kubernetes.io/docs/setup/best-practices/certificates/).
```
Comment (sftim) [inline content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-certs.md]: This is different from the style guide, and also the absolute URI should become site-relative.

How about:
```suggestion
[PKI certificates and requirements](/docs/setup/best-practices/certificates/) includes guidance on
setting up a cluster to use an external CA>
```

File: content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-certs.md [hunk 4]
```diff
@@ -21,9 +21,37 @@ You should be familiar with [PKI certificates and requirements in Kubernetes](/d
...
+
 ## Check certificate expiration
 
-`check-expiration` can be used to check certificate expiration.
+The `check-expiration` command can be used to check certificate expiration.
```
Comment (sftim) [inline content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-certs.md]: style nit:
```suggestion
You can use the `check-expiration` subcommand to check when certificates expire:
```
or
```suggestion
You can use kubeadm to check when certificates expire:
```

Also: mention alpha status?

Comment (sftim) [reply content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-certs.md]: Bear in mind that the overall page is marked stable:
```markdown
{{< feature-state for_k8s_version="v1.15" state="stable" >}}
```

Comment (neolit123) [reply content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-certs.md]: the command is already denoted as alpha in the shell example.

File: content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-certs.md (hunk 3 above)
Comment (neolit123) [reply content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-certs.md]: typo - `>` instead of `.` in suggestion.
i will update it.

**Comment by neolit123** [comment]
@sftim thanks for the review updated.

**Pull request merged**